### DISCUSSION TITLE
MAUI Firebase Push Notification not working on Android When App is closed

### DISCUSSION THREAD

**Discussion by vineethamanoj12** [shinyorg/shiny#1353] [Show and tell]
Hi,

As we are trying to integrate shiny Push notification with firebase on MAUI (.NET 7, Shiny 3.1.2) and its works fine on IOS while App closed/Open. But in Android Shiny works only when app is open. We followed the shiny Sample as 
[shiny/samples/Sample.Push.Maui at master · shinyorg/shiny (github.com)](https://github.com/shinyorg/shiny/tree/master/samples/Sample.Push.Maui) and we are not able to find any difference between our app and shiny Sample project.
So, anyone can suggest solution or anybody face same issue, please suggest a workaround to make our app to receive push notification when app is completely closed also.

**Comment by aritchie** [comment]
If you're sending the notification payload, the app doesn't need to be open.  The notification tray handles things.

**Reply by vineethamanoj12** [reply]
My Firebase message already contains a notification payload (Sample Included). (In fact, it's working perfectly with the Shiny Push notification demo app). However, our Android app does not display the notification if the app is completely closed. 
When I checked device logs its shows some warning messages, but I can't identify the problem please find attached device log

![DeviceLog](https://github.com/shinyorg/shiny/assets/132890425/2ca524c1-b5e9-4728-98ca-66b8429adc0e)
. 

 (If the app opens it works perfectly on Android, on IOS everything works perfectly). We have already compared settings with demo app & my app and all looks similar. How to resolve it? Thanks in advance.

data
       title: "Test Notification"
       body: "Test Body"
       page: "VerificationFeedbackPage"
to: "euaZqpYHSH…<DEVICE TOKEN"
content_available: True
notification
     title: " Test Notification "
     body: "Test Body"
click_action: "SHINY_PUSH_NOTIFICATION_CLICK"

**Reply by aritchie** [reply]
Please share a small sample on github and I can take a quick look

**Reply by vineethamanoj12** [reply]
Thank you for your support, we resolved this issue. In our app Notification is not showing because of some ThemeLoader code mentioned in our app. As we are loading our app theme by implementing IThemeLoader and we find the current window as 
  Window window = Activity. Window;
This window property is used to set StatusBarColor.
currentWindow.SetStatusBarColor(NavigationBarColor);   
When the app receives notification the above Activity.window  is not available and it leads to some internal error so our notification is not showing when app is closed . This we solved as, after checking the current window availability only we set StatusBarColor.
